enhancements/console/samples.md

```diff
@@ -0,0 +1,309 @@
+
+### Drawbacks
+
+Another extra (but specific) custom resource just for console samples.
+
+## Design Details
+
+An "Import from Git" example of how we have it today as part of the Builder Images annotation:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+spec:
+  title: Java
+  description: |
+    Build and run Java applications using Maven and OpenJDK.
+    Sample repository: https://github.com/jboss-openshift/openshift-quickstarts
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Serverless function
+  tags:
+  - java
+  - jboss
+  - openjdk
+  source:
+    type: GitImport
+    gitimport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+```
+
+An "Import container image" example:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: minimal-ubi-container
+spec:
+  title: Minimal UBI container
+  description: |
+    The Red Hat Universal Base Image is free to deploy on Red Hat or non-Red Hat platforms
+    and freely redistributable...
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Empty container
+  source:
+    type: ContainerImport
+    containerimport:
+      image: registry.access.redhat.com/ubi8/ubi-minimal:8.8-860
+```
+
+### Localization
+
+The `ConsoleSample` resources provide the same optional localization annotations then `ConsoleQuickStarts`.
+
+The `console.openshift.io/name: explore-serverless` label specifies the name of the sample. This should be consistent across all translations of a given sample,
+as it allows us to avoid showing the same Quick Start content in multiple languages.
+
+If a sample is not translated into a user'ss language, the english version will be shown.
+
+See the [internationalization](https://github.com/openshift/enhancements/blob/master/enhancements/console/internationalization.md) and [QuickStart enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/console/quick-starts.md) for more information.
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+  annotations:
+    console.openshift.io/name: java-sample # optional, metadata.name is also fine
+spec:
+  title: Java sample
+  description: |
+    Build and run Java applications using Maven and OpenJDK.
+    Sample repository: https://github.com/jboss-openshift/openshift-quickstarts
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Serverless function
+  tags:
+  - java
+  - jboss
+  - openjdk
+  source:
+    type: GitImport
+    gitimport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+---
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-samples-de
+  annotations:
+    console.openshift.io/lang: de
+    console.openshift.io/name: java-sample # same as annotation or metadata.name above
```

Comment:
Shouldn't these be labels rather than annotations to be consistent with quick starts? It would also let us use a label selector to only get languages we need on the frontend.

Reply:
:100:! Don't know why I wrote annotations :man_shrugging: